packages/next/src/lib/load-custom-routes.ts

```diff
@@ -138,7 +139,13 @@ export function checkCustomRoutes(
   let hadInvalidHas = false
   let hadInvalidMissing = false
 
-  const allowedKeys = new Set<string>(['source', 'locale', 'has', 'missing'])
+  const allowedKeys = new Set<string>([
+    'source',
+    'originalSource',
```

Comment:
We shouldn't pass `originalSource` on the route itself otherwise users could accidentally pass it in themselves when they see route validation errors.

Reply:
I moved this back to keeping a separate collection of routes mapped to a value so we don't have to modify the route itself: [`0ae9b61` (#47980)](https://github.com/vercel/next.js/pull/47980/commits/0ae9b61eb0c6553d5aa818be1e63a7a056518b22)